Packet++/src/HttpLayer.cpp

```diff
@@ -855,14 +873,31 @@ HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data,
 		return HttpResponseStatusCode::HttpStatusCodeUnknown;
 	}
 
-	std::string codeString = std::string(data + 9, 3);
+	const std::string codeString = std::string(data + 9, 3);
 
 	if(codeString.empty() || (std::find_if(codeString.begin(), codeString.end(), [](unsigned char c){ return !std::isdigit(c); }) != codeString.end()))
 	{
 		return HttpResponseStatusCode::HttpStatusCodeUnknown;
 	}
 
-	return HttpResponseStatusCode(std::stoi(codeString));
+	constexpr size_t messageOffset = 13;  // expect "HTTP/x.y XXX YYY", YYY starts from 13
+	size_t offset = messageOffset;
+	bool isMessageFound = false;
+	while(offset < dataLen)
+	{
+		if(data[offset] == '\n')
+		{
+			isMessageFound = true;
+			break;
+		}
+		offset++;
+	}
+	std::string messageString = isMessageFound ? std::string(data + messageOffset, offset - messageOffset) : "";
```

Comment:
if `isMessageFound == false` I think this is an invalid HTTP message and we can return `HttpResponseStatusCode::HttpStatusCodeUnknown`, no? 🤔

Reply:
yeah... I just forgot a valid first line should always close with \n